src/lib/plugin_apis/fs.api

```diff
@@ -361,6 +361,62 @@ gchar* bd_fs_get_mountpoint (const gchar *device, GError **error);
  */
 gboolean bd_fs_resize (const gchar *device, guint64 new_size, GError **error);
 
+typedef enum {
+    BD_FS_NO_SUPPORT,
+    BD_FS_OFFLINE,
+    BD_FS_OFFLINEGROW,
+    BD_FS_ONLINE,
+    BD_FS_ONLINEGROW,
+    BD_FS_ONLINE_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINEGROW,
+} BDFsOnlineOffline;
+
+/**
+ * bd_fs_can_resize:
+ * @type: the filesystem type to be tested for installed resize support
+ * @mode: (out): how the filesystem allows resizing (e.g. online, offline, growing or not at all)
+ * @required_utility: (out): the utility binary which is required for resizing (if any)
```

Comment:
Hello, thanks for the review.
My idea was to only return the utility binary if it is required (i.e. empty for vfat) so that if FALSE is returned and no error is set, it is clear that only the binary is missing and could be installed with PackageKit etc.

I thought since the default is (transfer none) and only static strings needed it would be ok like that. But yes, ownership transfer is what people would expect and return NULL instead of an empty string for vfat.